Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.cpp

```diff
@@ -0,0 +1,47 @@
+#include "fontcontrolswidgetsform.h"
+#include "ui_fontcontrolswidgetsform.h"
+
+FontControlsWidgetsForm::FontControlsWidgetsForm(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::FontControlsWidgetsForm)
+{
+    ui->setupUi(this);
+    connect(ui->lineEdit, &QLineEdit::textEdited, this, &FontControlsWidgetsForm::lineEdit_textChanged);
+    connect(ui->fontComboBox, &QFontComboBox::currentFontChanged, this, &FontControlsWidgetsForm::fontComboBox_currentFontChanged);
+    connect(ui->pushButton, &QPushButton::clicked, this, &FontControlsWidgetsForm::pushButton_clicked);
+}
+
+FontControlsWidgetsForm::~FontControlsWidgetsForm()
+{
+    delete ui;
+}
+
+const QString FontControlsWidgetsForm::text()
+{
+    return ui->lineEdit->text();
+}
+
+void FontControlsWidgetsForm::lineEdit_textChanged(const QString &arg1)
+{
+    emit textChanged();
+}
+
+void FontControlsWidgetsForm::setText(const QString &text)
+{
+    ui->lineEdit->setText(text);
+    emit textChanged();
+}
+
+const QString FontControlsWidgetsForm::font()
+{
+    return ui->fontComboBox->currentFont().family();
+}
+
+void FontControlsWidgetsForm::closeEvent(QCloseEvent *event) {
```

Comment:
Inconsistent formatting: you inserted a newline before '{' in most other methods (which is indeed more common).

You *could* grab the _clang-format from e.g. KDABViewer and run clang-format over the whole code, for consistent formatting.